#! /bin/sh /usr/share/dpatch/dpatch-run
## 96-jquery_compat.dpatch by Markus Frosch <lazyfrosch@debian.org>
##
## DP: Bug#831332 icinga-cgi-bin: fix menu hiding (jquery 1.11 incompatibility)
## DP:
## DP: icinga-cgi-bin in Debian Sid, Ubuntu Yakkety (and Ubuntu Xenial) is
## DP: affected by an upstream bug where the left-hand menu in the Classic
## DP: icinga web interface disappears immediately after the page finishes
## DP: loading. Upstream has provided a fix and in my testing this resolves the
## DP: issue (which is due to current code incompatibilty with jQuery versions
## DP: in Debian and Ubuntu).

@DPATCH@
--- icinga-1.13.3.orig/html/js/menu.js
+++ icinga-1.13.3/html/js/menu.js
@@ -9,11 +9,14 @@
 
 		elem.prepend( img )
 			.css( "cursor", "pointer" )
-			.toggle( function() {
-				img.attr('src', 'images/menu_more.gif');
-				list.slideToggle("slow");
-			}, function() {
-				img.attr('src', 'images/menu_less.gif');
+			.click( function() {
+				if($(this).hasClass("collapsed")) {
+					img.attr('src', 'images/menu_less.gif');
+					$(this).removeClass("collapsed");
+				} else {
+					img.attr('src', 'images/menu_more.gif');
+					$(this).addClass("collapsed");
+				}
 				list.slideToggle("slow");
 		});
